test-requirements.txt

```diff
@@ -29,3 +29,4 @@ ipdb~=0.11
 pip<10
 wheel~=0.29
 testfixtures~=5.3.1
+regex
```

Comment:
any special reason why this extra module dependency?
what are the shortcomings of the default `re` module and what benefits do we get by using this module?

Reply:
In the parser - we iterate through match objects present on the same line. Each Regex string have been has the  tags such as `open` , `close` and `content`. For eg:
```python
JINJA_VAR_PATTERN = (
    r'(?P<open>{{)(?P<content>.*?)(?P<close>}})')
```

Now on a particular line there would be cases where we can have more than one `match object` , for eg:
```yaml
x = {{ var1 }} + {{ var2 }}
```  
If we use `re` module we are thrown an error because both the matched strings have same `tags` ( open, close and content). `regex` module can overcome this situation.

Reply:
Can't replicate the issue: https://ideone.com/KZK1nF

Reply:
Please have a look at https://ideone.com/9LlsE0

Reply:
Right, so document that your parser specifically requires the regex import for branch reset functionality somewhere in the docstring

Reply:
Comment line added for [regex](https://github.com/Naveenaidu/coala/pull/2/files#diff-8e44aad653930aec57884df55b514badR4)